### PR TITLE
[catalog] ensure CKAN dirs exist

### PR DESCRIPTION
**Pull request by adborden**
When installing on a new EC2 instance in sandbox, seeing errors that
directories don't exist. Ensure they are created with the correct permissions

### CODE REVIEW

**Comment by adborden** [comment]
Rebase to pick up Snyk fix.

**Comment by adborden** [comment]
Adding a work around for https://github.com/GSA/catalog-app/issues/76, standby.

**Comment by adborden** [comment]
Okay, I think this is ready for review.

If this comes back in development, I think that just creating the file `touch /var/tmp/ckan/dynamic_menu/menu.json` might work, but we should really fix this in the extension.

File: ansible/roles/software/ckan/catalog/ckan-app/tasks/main.yml [hunk 1]
```diff
@@ -11,6 +11,20 @@
...
+
+- name: create CKAN temp directory
+  file: path=/var/tmp/ckan owner=www-data group=www-data mode=0755 state=directory
+
+- name: create CKAN storage directory
+  file: path=/var/lib/ckan owner=www-data group=www-data mode=0755 state=directory
+
+# Work-around https://github.com/GSA/catalog-app/issues/76
+- name: pre-populate menu.json for ckanext-datagovtheme
+  copy: src=menu.json dest=/var/tmp/ckan/dynamic_menu/menu.json owner=www-data group=www-data mode=0644
```
Comment (FuhuXia) [inline ansible/roles/software/ckan/catalog/ckan-app/tasks/main.yml]: Can we set this file with an old timestamp, so the app will try to fetch up-to-date content of menu.json.

**Comment by adborden** [comment]
Quoting Fuhu Xia (2020-03-19 07:24:14)
> +# Work-around https://github.com/GSA/catalog-app/issues/76
> +- name: pre-populate menu.json for ckanext-datagovtheme
> +  copy: src=menu.json dest=/var/tmp/ckan/dynamic_menu/menu.json owner=www-data group=www-data mode=0644
> 
> Can we set this file with an old timestamp, so the app will try to fetch up-to-date content of menu.json.  

Good idea!


--
Aaron D Borden
Human and hacker

**Comment by adborden** [comment]
Ok, ready for another :eyes:

**Review by jbrown-xentity — APPROVED** [review]
Looks great!

**Pull request merged**